src/ModelContextProtocol/Configuration/McpServerBuilderExtensions.cs

```diff
@@ -242,6 +243,119 @@ where t.GetCustomAttribute<McpServerPromptTypeAttribute>() is not null
     }
     #endregion
 
+    #region WithResources
+
+    private static McpServerResource ToResource(
+        this IFileInfo fileInfo)
+    {
+        Throw.IfNullOrWhiteSpace(fileInfo.PhysicalPath);
+
+        return new()
+        {
+            ProtocolResource = new()
+            {
+                Uri = fileInfo.PhysicalPath,
+                Name = fileInfo.Name,
+            },
+        };
+    }
+
+    private static IMcpServerBuilder AddResource(
+        this IMcpServerBuilder builder,
+        McpServerResource resource)
+    {
+        builder.Services.Configure<McpServerOptions>(s =>
+        {
+            var capabilities = s.Capabilities ??= new();
+            var resources = capabilities.Resources ??= new();
+            var collection = resources.ResourceCollection ??= [];
+            collection.Add(resource);
+        });
+
+        return builder;
+    }
+
+    private static IMcpServerBuilder AddResources(
+        this IMcpServerBuilder builder,
+        IEnumerable<McpServerResource> resources)
+    {
+        foreach (var resource in resources)
+        {
+            builder = builder.AddResource(resource);
+        }
+        return builder;
+    }
+
+    /// <summary>
+    /// Adds a resource to the server's capabilities.
+    /// </summary>
+    /// <param name="builder">The builder instance.</param>
+    /// <param name="resource">The resource to add.</param>
+    /// <returns>The <see cref="IMcpServerBuilder"/> instance.</returns>
+    public static IMcpServerBuilder WithResource(
+        this IMcpServerBuilder builder,
+        McpServerResource resource)
+    {
+        Throw.IfNull(builder);
+        Throw.IfNull(resource);
+
+        return builder.AddResource(resource);
+    }
+
+    /// <summary>
+    /// Adds a resource to the server's capabilities.
+    /// </summary>
+    /// <param name="builder">The builder instance.</param>
+    /// <param name="fileInfo">The file info of the resource.</param>
+    /// <returns>The <see cref="IMcpServerBuilder"/> instance.</returns>
+    public static IMcpServerBuilder WithResource(
+        this IMcpServerBuilder builder,
+        IFileInfo fileInfo)
```

Comment:
Isn't it critically important to have a MIME type for a resource? Otherwise the client doesn't know how to interpret the bytes. And IFileInfo doesn't provide any information about the media type (for a limited set of known file extensions we might be able to infer it, but it would require a hardcoded list and would likely fall off a cliff).